src/runtime/handle.ts

```diff
@@ -361,7 +387,7 @@ class Cursor {
    * when the cursor has completed reading the collection.
    */
   async next() {
-    const data = await this._parent.storage.cursorNext(this._cursorId);
+    const data = await this._parent.getStorage().cursorNext(this._cursorId);
```

Comment:
You can consider changing getStorage to be a javascript getter, which would allow you to revert this lines to their original form.

```
  get storage(): Readonly<SingletonStore> {
    return this.storage;
  }
```

Reply:
Changed! I ended up having to change the field to `_storage` so there are no duplicates.